onnxruntime/core/optimizer/insert_cast_transformer.cc

```diff
@@ -70,7 +160,7 @@ Status ForceSingleNodeCPUFloat16ToFloat32(onnxruntime::Graph& graph) {
   }
 
   for (auto& node : graph.Nodes()) {
-    if (node.OpType() != "Cast" && IsSingleInputNodeFloat16Node(node)) {
+    if (IsIsolatedFp16Node(node, graph)) {
```

Comment:
1) I am guessing clearing the execution provider here means the downstream logic will force the node to fp32 on the CPU EP and insert casts around it ? Can we add a comment here stating what really is the intent of just clearing the EP for the node ? It might help someone stumbling on this method.

2) Should `IsIsolatedFp16Node()` be renamed to `IsIsolatedFp16NodeOnCpu()` to be more explicit about the fact that this kind of forcing is only done for nodes assigned to the CPU EP ?

Reply:
1. Added comment
2. Renamed as suggested